src/utils/consoleColor.lua

```diff
@@ -0,0 +1,35 @@
+local ANSI_COLOR_TEMPLATE = "\x1b[%s;%s;%sm"
+local DEFAULT_BG_CODE = 49
+
+local RESET_CODE = ANSI_COLOR_TEMPLATE:format(0, 0, 0)
+
+--- Creates a function which takes in text and colors it according to the arguments.
+---@param modifier number 0 means normal, 1 means bold and 2 means dimmed
+---@param colorCode number ANSI escape color code
+---@param bgColorCode number|nil Optional background color code
+---@return fun(text:string|nil):string textFn
+local function createColoredTextFn(modifier, colorCode, bgColorCode)
+	return function(text)
+		text = text or ""
+		bgColorCode = bgColorCode or DEFAULT_BG_CODE
+
+		return ANSI_COLOR_TEMPLATE:format(
+			tostring(modifier),
+			tostring(colorCode),
+			tostring(bgColorCode)
+		) .. text .. RESET_CODE
```

Comment:
Do you need to use both `string.format` and string concatenation here? Can you just add `%s%s` to the end of `ANSI_COLOR_TEMPLATE`?

In fact, you can remove `RESET_CODE` entirely and simply make `ANSI_COLOR_TEMPLATE` include it.